test/testenv/s3utils.go

```diff
@@ -226,3 +229,74 @@ func UploadFilesToS3(testS3Bucket string, s3TestDir string, applist []string, do
 	}
 	return uploadedFiles, nil
 }
+
+// DisableAppsToS3 download apps from S3, untar them, modify their conf file to disable them, re-tar and upload the disabled version to S3
+func DisableAppsToS3(testDataS3Bucket string, s3AppDir string, downloadDir string, appFileList []string, s3TestDir string) ([]string, error) {
+
+	// Download apps tar files
+	DownloadFilesFromS3(testDataS3Bucket, s3AppDir, downloadDir, appFileList)
```

Comment:
List of apps to be disabled is provided through appFileList parameter, so all the apps in that list would be disabled. That list has to be created correctly in the tests before calling DisableAppsToS3.

Also about the apps files, it would look like this:
<xxappfwxx> folder:
    - disabled apps folder (containing tgz of disabled apps, for ex: 'microsoft-azure-add-on-for-splunk.tgz'
    - original apps tgz files for ex: 'microsoft-azure-add-on-for-splunk.tgz'
    - untarred_apps folder:
         - folders with name of the tgz file minus extension, for ex: microsoft-azure-add-on-for-splunk
              - inside each of these folders, the untarred app root folder, for ex:'TA-MS-AAD'

When uploading to S3, it would be like this:
<xxappfwxx> folder, 
    - 'disabled_apps' folder inside, containing the disabled tgz files

Reply:
For the download would it not be better to just upload to the required s3 location that the appframework spec is pointing to.

Would avoid extra downloads

Reply:
Like @tpereirasplunk mentioned the apps should be present on the local directory from the test download ?